app/src/main/java/com/parishod/watomatic/fragment/SettingsFragment.java

```diff
@@ -0,0 +1,39 @@
+package com.parishod.watomatic.fragment;
+
+import android.os.Bundle;
+import android.util.Log;
+
+import androidx.preference.ListPreference;
+import androidx.preference.Preference;
+import androidx.preference.PreferenceFragmentCompat;
+
+import com.parishod.watomatic.R;
+import com.parishod.watomatic.activity.main.MainActivity;
+import com.parishod.watomatic.model.preferences.PreferencesManager;
+
+public class SettingsFragment extends PreferenceFragmentCompat {
+    private ListPreference languagePref;
+
+    @Override
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+        setPreferencesFromResource(R.xml.fragment_settings, rootKey);
+
+        //Set fragment title
+        getActivity().setTitle("Settings");
+
+        ((MainActivity)getActivity()).showHideBackButton(true);
```

Comment:
We can also avoid these hacks if we move to separate activity if my understanding is right.

Reply:
Or I should ask, what does this do? I see this in MainFragment too